### PR TITLE
Amanda lane

### PR DESCRIPTION
**Pull request by amlane**
### Day 1: Get `print8.ls8` running

- [x] Inventory what is here
- [x] Implement the `CPU` constructor
- [x] Add RAM functions `ram_read()` and `ram_write()`
- [x] Implement the core of `run()`
- [x] Implement the `HLT` instruction handler
- [x] Add the `LDI` instruction
- [x] Add the `PRN` instruction

### CODE REVIEW

**Comment by amlane** [comment]
### Day 2: Add the ability to load files dynamically, get `mult.ls8` running

- [x] Un-hardcode the machine code
- [x] Implement the `load()` function to load an `.ls8` file given the filename
      passed in as an argument
- [x] Implement a Multiply instruction (run `mult8.ls8`)

**Comment by amlane** [comment]
### Day 3: Stack

- [x] Implement the System Stack and be able to run the `stack.ls8` program

**Comment by amlane** [comment]
### Day 4: Get `call.ls8` running

- [x] Implement the CALL and RET instructions
- [x] Implement Subroutine Calls and be able to run the `call.ls8` program

File: ls8/cpu.py [hunk 1]
```diff
@@ -62,4 +99,64 @@ def trace(self):
...
+            elif IR == PRN:
+                data = self.ram_read(self.pc + 1)
+                # print the data
+                print(self.reg[data])
+                # increment the PC by 2 to skip the argument
+            elif IR == MUL or IR == ADD:
+                op = IR
+                reg_a = self.ram_read(self.pc + 1)
+                reg_b = self.ram_read(self.pc + 2)
+                self.alu(op, reg_a, reg_b)
```
Comment (BrannanC) [inline ls8/cpu.py]: Maybe the assignment to `op` isn't needed, `self.alu(IR, reg_a, reg_b)`, but I get why you would remind yourself that it's the current operation

File: ls8/cpu.py [hunk 2]
```diff
@@ -2,41 +2,78 @@
...
+MUL = 162
+PUSH = 69
+POP = 70
+CALL = 80
+ADD = 160
+RET = 17
+SUB = None
+DIV = None
+
+SP = 7  # Stack Pointer
```
Comment (BrannanC) [inline ls8/cpu.py:25]: Yep, this is really good. I think there's other instances where the spec designates a specific register for something? Those eat into your 8 registers, but you could also write a program that completely ignores the SP designation and uses that register for whatever you want.

**Review by BrannanC — COMMENTED** [review]
So this is extremely clean, and way more organized than mine 😂 
I definitely don't suggest refactoring your code at all, but keep in mind that assigning variables is kind of cheating. "Variables" are stored in registers or at specific locations in memory. I know I'm being pedantic, but just trying to reinforce that this code represents physical components.

**Pull request merged**